src/test/java/se/sundsvall/parkingpermit/businesslogic/worker/execution/OrderCardTaskWorkerTest.java

```diff
@@ -0,0 +1,129 @@
+import static se.sundsvall.parkingpermit.Constants.CAMUNDA_VARIABLE_REQUEST_ID;
+
+@ExtendWith(MockitoExtension.class)
+class OrderCardTaskWorkerTest {
+
+	private static final String REQUEST_ID = "RequestId";
+	private static final long ERRAND_ID = 123L;
+	private static final String VARIABLE_CASE_NUMBER = "caseNumber";
+	private static final String VARIABLE_REQUEST_ID = "requestId";
+	private static final String QUEUE_NEW_CARD = "NyttKortNyPerson";
+	private static final String QUEUE_REPLACEMENT_CARD = "NyttKortBefintligPerson";
+	private static final String QUEUE_ANTI_THEFT_AND_REPLACEMENT_CARD = "StöldspärraSamtSkapaNyttKort";
+	private static final String CASEDATA_STATUS_DECISION_EXECUTED = "Beslut verkställt";
+
+	@Mock
+	private CaseDataClient caseDataClientMock;
+
+	@Mock
+	private RpaService rpaServiceMock;
+
+	@Mock
+	private ExternalTask externalTaskMock;
+
+	@Mock
+	private ExternalTaskService externalTaskServiceMock;
+
+	@Mock
+	private FailureHandler failureHandlerMock;
+
+	@Captor
+	private ArgumentCaptor<List<StatusDTO>> statusesDTOArgumentCaptor;
+
+	@InjectMocks
+	private OrderCardTaskWorker worker;
+
+	@ParameterizedTest
+	@MethodSource("orderCardTypeArguments")
+	void execute(CaseTypeEnum caseType, List<String> expectedQueueNames) {
+		//Arrange
+		final var errand = new ErrandDTO().id(ERRAND_ID).caseType(caseType);
+		when(externalTaskMock.getVariable(VARIABLE_REQUEST_ID)).thenReturn(REQUEST_ID);
+		when(caseDataClientMock.getErrandById(ERRAND_ID)).thenReturn(errand);
+		when(externalTaskMock.getVariable(VARIABLE_CASE_NUMBER)).thenReturn(ERRAND_ID);
+		// Act
+		worker.execute(externalTaskMock, externalTaskServiceMock);
+
+		// Assert and verify
+		verify(externalTaskMock).getVariable(CAMUNDA_VARIABLE_REQUEST_ID);
+		verify(rpaServiceMock).addQueueItems(expectedQueueNames, ERRAND_ID);
+		verify(caseDataClientMock).putStatus(eq(ERRAND_ID), statusesDTOArgumentCaptor.capture());
+		verify(externalTaskServiceMock).complete(externalTaskMock);
+		assertThat(statusesDTOArgumentCaptor.getValue()).hasSize(1)
+			.extracting(StatusDTO::getStatusType, StatusDTO::getDescription)
+			.containsExactly(
+				tuple(CASEDATA_STATUS_DECISION_EXECUTED, CASEDATA_STATUS_DECISION_EXECUTED));
+		assertThat(statusesDTOArgumentCaptor.getValue().get(0).getDateTime()).isCloseTo(OffsetDateTime.now(), within(1, SECONDS));
+		verifyNoInteractions(failureHandlerMock);
+	}
+
+	@Test
+	void executeThrowsException() {
+		// Arrange
+		final var errand = new ErrandDTO().id(ERRAND_ID).caseType(PARKING_PERMIT);
+		when(externalTaskMock.getVariable(VARIABLE_REQUEST_ID)).thenReturn(REQUEST_ID);
+		when(caseDataClientMock.getErrandById(ERRAND_ID)).thenReturn(errand);
+		when(externalTaskMock.getVariable(VARIABLE_CASE_NUMBER)).thenReturn(ERRAND_ID);
+		final var thrownException = new EngineException("TestException", new RestException("message", "type", 1));
+
+		// Mock
+		doThrow(thrownException).when(externalTaskServiceMock).complete(any());
```

Comment:
Change to throw exception from rpaServiceMock instead, to verify that complete-method of externalTaskServiceMock is not called when exception occur in handler?